### PR TITLE
Use std futures, upgrade wasm-bindgen to latest

### PR DESCRIPTION
**Pull request by zimond**
Note: this PR is not ready as I found the test is using a field in `Model` which does not happen to be there on master branch

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,13 +18,12 @@ build = "build.rs"
 version_check = "^0.9.1"
 
 [dev-dependencies]
-wasm-bindgen-test = "0.2.50" # NOTE: keep in sync with wasm-bindgen version
+wasm-bindgen-test = "0.3.5" # NOTE: keep in sync with wasm-bindgen version
```
Comment (MartinKavik) [inline Cargo.toml]: remove `# NOTE: keep in sync with wasm-bindgen version` please, it looks that it isn't required anymore

File: Cargo.toml [hunk 2]
```diff
@@ -33,10 +32,11 @@ serde = { version = "^1.0.92", features = ['derive'] }
 serde_json = "^1.0.39"
 # NOTE: keep in sync with wasm-bindgen-test version
```
Comment (MartinKavik) [inline Cargo.toml]: remove `# NOTE: keep in sync with wasm-bindgen-test version` please, it looks that it isn't required anymore

File: src/fetch.rs [hunk 3]
```diff
@@ -436,196 +436,161 @@ impl Request {
     ///        .fetch(Msg::Fetched)
     ///}
     /// ```
-    pub fn fetch<U>(self, f: impl FnOnce(FetchObject<()>) -> U) -> impl Future<Item = U, Error = U>
+    pub async fn fetch<U>(self, f: impl FnOnce(FetchObject<()>) -> U) -> Result<U, U>
     where
         U: 'static,
     {
         // @TODO: once await/async stabilized, refactor
```
Comment (MartinKavik) [inline src/fetch.rs]: remove this comment please (on more places)

File: Cargo.toml (hunk 2 above)
Comment (zimond) [reply Cargo.toml]: yeah... I just want to clear these things after the test passes

Comment (MartinKavik) [reply Cargo.toml]: Ok, let me know, once the PR is ready for code review :)

**Comment by zimond** [comment]
@MartinKavik Hey, current github action will fail on clippy, as the codebase has many clippy warnings. Do you want me to clear them in this PR?

**Comment by MartinKavik** [comment]
@zimond yes, please, `master` should contain only formatted and linted code. I often create a commit like `fix: fmt + clippy`.

**Comment by zimond** [comment]
@MartinKavik ok, there's quite a lot. So I need some time

**Comment by zimond** [comment]
Currently CI fails with `gloo-timer` not publishing a new version with futures 0.3 changes.

**Comment by MartinKavik** [comment]
@zimond We can't have unpublished dependency in `cargo.toml` because it's not possible to release a version with it. So I think we need to wait.

**Comment by zimond** [comment]
@MartinKavik The problem here is that one of seed's examples is using gloo-timers' `TimeoutFuture`. Maybe we can just change the example and make this happen earlier

**Comment by MartinKavik** [comment]
@zimond Try it and add comments with `todo` that we should rewrite it once `gloo` supports `futures 0.3`, please.

**Comment by zimond** [comment]
@MartinKavik it's updated.

**Comment by MartinKavik** [comment]
_Note_: code review / merge blocked by https://github.com/zimond/seed/pull/1

**Comment by MartinKavik** [comment]
I think we can remove old `orders.rs` and `vdom.rs` files and it's good to go.

Sorry for conflicts with the `master` - we are preparing Seed code base for attribute API and virtual DOM updates. So thank you for update and I'm glad you took it because I don't have much experience with async/await in Rust yet.

@David-OConnor Once the files are removed, I'll do a final code review and probably merge it. Is it ok with you?

**Comment by zimond** [comment]
@MartinKavik hey the files are removed

**Comment by zimond** [comment]
Is there anything blocking?

**Comment by David-OConnor** [comment]
Looks good to me

**Comment by David-OConnor** [comment]
Would you mind updating the merge conflicts? Due to a recent merged PR updating examples. Will merge this after.

**Comment by zimond** [comment]
.......

**Comment by zimond** [comment]
Note: this PR takes so long that a new stable rust version is established, and now clippy's const fn suggestion is landed. So many clippy warnings that are previously not triggered are thrown now.

I decide to not correct them as that goes beyond the scope of this PR

**Comment by David-OConnor** [comment]
https://github.com/seed-rs/seed/pull/321

**Comment by zimond** [comment]
closes #284

**Pull request merged**